### PR TITLE
Issue53

### PR DESCRIPTION
**Pull request by zavodnyuk**
connect to #53

### CODE REVIEW

**Comment by roma-ilnitsky** [comment]
@zavodnyuk 
1. В класі `MailReminderAboutEventService` в методі `sendEmails` замість `userWhoGoToNearestEvent` краще використоти user, оскільки метод просто відправляє листи, він нічого не знає про юзерів
2. Поправити конфлікти із мастером. Зараз у нас в проперті всі зміні називаються через крапку, тому `cron.remindAboutEvent` треба буде перейменувати на `cron.remind.about.event` і те саме із `cron.start_sending `

**Comment by zavodnyuk** [comment]
@vaa25 Thanks for your rewiev.
@AndriyBaibak 2 hours

**Comment by vaa25** [comment]
+ PASSED
@AndriyBaibak 10 min

**Comment by romach** [comment]
@AndriyBaibak time spent: 30 minutes on code view

**Comment by zavodnyuk** [comment]
@AndriyBaibak 4 hours
@alex-anakin refactored

**Comment by vaa25** [comment]
@zavodnyuk 
1. All above.
2. MyBatisEventService.getEventsByDate() used only in test. Dead code.
3. MyBatisVisitLogService has unused imports
4. VisitLogService has unused import
5. Has changed changelogs
6. etc...

Although this pr has my `PASSED` but merge it into maser now must not.

**Comment by vaa25** [comment]
@AndriyBaibak 1 hour

**Comment by vaa25** [comment]
@zavodnyuk try to compare your branch with master before pr please

**Comment by zavodnyuk** [comment]
@AndriyBaibak 4 hours

**Comment by zavodnyuk** [comment]
@AndriyBaibak 1h 30 min

**Comment by romach** [comment]
@zavodnyuk need merge with master

**Comment by romach** [comment]
@zavodnyuk I get
```
org.springframework.beans.factory.BeanCreationException: Could not autowire field: private int org.itevents.service.sendmail.MailEventRemin
derService.daysTillEvent; nested exception is java.lang.IllegalArgumentException: Could not resolve placeholder 'days.till.event.to.remind' in string v
alue "${days.till.event.to.remind}"
                                                                  
                Caused by:                                        
                java.lang.IllegalArgumentException: Could not resolve placeholder 'days.till.event.to.remind' in string value "${days.till.event.to.rem
ind}"
```
while build

**Comment by zavodnyuk** [comment]
@romach ok, look to last commit. You need add property to local.proprties and it will be okey. 
Branch merged with master

File: restservice/src/test/java/org/itevents/service/sendmail/MailEventReminderServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    @Before
+    public void init(){
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void shouldGetUsersByEventDateAndSendMail() throws Exception {
+
+        List<Event> events = new ArrayList<Event>();
```
Comment (romach) [inline restservice/src/test/java/org/itevents/service/sendmail/MailEventReminderServiceTest.java]: @zavodnyuk Can be replaced with `ArrayList<>`

File: restservice/src/test/java/org/itevents/service/sendmail/MailEventReminderServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        events.add(BuilderUtil.buildEventJava());
+        when(eventDao.getEventsByDate(any(Date.class))).thenReturn(events);
+
+        List<User> user = new ArrayList<User>();
```
Comment (romach) [inline restservice/src/test/java/org/itevents/service/sendmail/MailEventReminderServiceTest.java]: @zavodnyuk Can be replaced with `ArrayList<>`

File: restservice/src/test/java/org/itevents/service/sendmail/MailEventReminderServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        user.add(BuilderUtil.buildSubscriberTest());
+        when(userDao.getUsersByEvent(any(Event.class))).thenReturn(user);
+
+        doNothing().when(mailService).sendMail(anyString(), anyString());
```
Comment (romach) [inline restservice/src/test/java/org/itevents/service/sendmail/MailEventReminderServiceTest.java]: @zavodnyuk Why do you need this `doNothing()`? `mailService` is mock. So it doesn't do any actions.

File: restservice/src/test/java/org/itevents/service/sendmail/MailEventReminderServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+        mailReminderAboutEventService.remind();
+
+        verify(eventDao).getEventsByDate(any(Date.class));
+        verify(userDao, times(events.size())).getUsersByEvent(any((Event.class)));
+        verify(mailService, times(user.size())).sendMail(anyString(), anyString());
```
Comment (romach) [inline restservice/src/test/java/org/itevents/service/sendmail/MailEventReminderServiceTest.java]: @zavodnyuk It is bad practice to use `any...` in `verify()`

File: restservice/src/test/java/org/itevents/service/sendmail/MailEventReminderServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        List<User> user = new ArrayList<User>();
+        user.add(BuilderUtil.buildSubscriberTest());
+        when(userDao.getUsersByEvent(any(Event.class))).thenReturn(user);
+
+        doNothing().when(mailService).sendMail(anyString(), anyString());
+
+        mailReminderAboutEventService.remind();
+
+        verify(eventDao).getEventsByDate(any(Date.class));
+        verify(userDao, times(events.size())).getUsersByEvent(any((Event.class)));
```
Comment (romach) [inline restservice/src/test/java/org/itevents/service/sendmail/MailEventReminderServiceTest.java]: @zavodnyuk unnecessary parentheses `((Event.class))`

File: restservice/src/main/java/org/itevents/service/scheduler/MailSenderScheduler.java [hunk 6]
```diff
@@ -15,8 +17,25 @@
...
+    @Scheduled(cron = "${cron.remind.about.event}")
+    public void startRemind(){
+        try {
+            eventReminderService.remind();
+        }catch (BuildMailException e){
+            /*
+            * @TODO: need to send mail to Admin with information, that mail to user wasn't built and sent.
+            * issue53
+            * https://github.com/JuniorsJava/itevents/issues/53
+            */
```
Comment (romach) [inline restservice/src/main/java/org/itevents/service/scheduler/MailSenderScheduler.java]: You also need do log this exception

**Comment by romach** [comment]
@AndriyBaibak 1 hour spent

**Comment by zavodnyuk** [comment]
@AndriyBaibak 30 min
@romach All advices refactored

**Pull request merged**